test_files/functions/functions.ts

```diff
@@ -34,6 +34,20 @@ TestSplat3(1, 2);
 function defaultBeforeRequired(x = 1, y: number, ...z: any[]) {}
 defaultBeforeRequired(undefined, 2, 'h', 3);
 
+// The array reference below happens on the template parameter constraint, not on the parameter
+// itself. Make sure tsickle unwraps the right type.
+function templatedBoundRestArg<T extends string[]>(...str: T) {
```

Comment:
I tried this on the playground and got:
ERROR(1,52): : A rest parameter must be of an array type.

Reply:
Maybe it's a version issue? This works (and we do also check for diagnostics in these tests): 

http://www.typescriptlang.org/play/#src=function%20templatedBoundRestArg%3CT%20extends%20string%5B%5D%3E(...str%3A%20T)%20%7B%20%7D%0D%0AtemplatedBoundRestArg('a'%2C%20'b')%3B

Reply:
Wow it fails on TS29, and works on TS31!  (I was looking at this to try to understand what the behavior is...)